### PR TITLE
Adding __version__, fix for issue #381

### PR DESCRIPTION
**Pull request by nandahkrishna**
Fixes issue #381 by adding __version__ to init.py, using code similar to the docs. Please do tell me if there's something I have to change here.

### CODE REVIEW

**Comment by nandahkrishna** [comment]
The Travis CI shows failure on MacOS with benchmarks and optional dependencies (Build 1270.5), but passes the rest.

File: rx/__init__.py [hunk 1]
```diff
@@ -3,8 +3,23 @@
 from asyncio.futures import Future as _Future
 from typing import Iterable, Callable, Any, Optional, Union
 
+import os
```
Comment (jcafhe) [inline rx/__init__.py]: Maybe modules from the standard library should be imported first, as suggested by the [python style guide](https://www.python.org/dev/peps/pep-0008/#imports).

**Comment by jcafhe** [comment]
> The Travis CI shows failure on MacOS with benchmarks and optional dependencies (Build 1270.5), but passes the rest.

Indeed, but don't worry, we have an issue with MacOs tests for now.

File: rx/__init__.py (hunk 1 above)
Comment (nandahkrishna) [reply rx/__init__.py]: Alright, I'll fix this.

**Comment by nandahkrishna** [comment]
Glad to contribute!

**Comment by MainRo** [comment]
isn't opening, reading and parsing a file when loading the rxpy module only to retrieve the version number overkill ? We do the same at different locations (setup.py and doc/conf.py) but not during runtime. I wonder if using bumpversion would not be easier to get the version number in all these locations ?

https://pypi.org/project/bumpversion/

**Comment by nandahkrishna** [comment]
>  I wonder if using bumpversion would not be easier to get the version number in all these locations ?
Certainly does seem like a nice way to go about this. Looking at it, I guess a couple of changes here and there would have to be made, like a new bumpversion.cfg and changes in the docs, etc.

I can go ahead and make the change to bump version if you'd prefer that.

**Comment by erikkemperman** [comment]
Agree it might not be something we want to do at runtime. It's nice to have all these properties in one place, which was the point of project.cfg, but not at any cost...

Maybe the version number could be fixed in the `__init__.py` and read from `setup.py` and `docs/conf.py`? Or, indeed, something like the tool you mention would probably work well.

**Comment by nandahkrishna** [comment]
Well, I think the bumpversion tool seems to be a better option, given that it is quite similar to the current project.cfg - lets you have all the information in the same place. I think I'll give that a try.

**Comment by nandahkrishna** [comment]
The project.cfg file has been removed in favour of .bumpversion.cfg which contains the same details or meta-data, but now supports version string updates using `bumpversion` or `bump2version`.

With the current version being `3.0.0-beta1`, the version has been split into 4 parts - major (3), minor(0), patch(0) and release (beta1).

Here are some examples on how the version string update can be done:
`bump2version release` changes it to `3.0.0-beta2` - this is up for review and the options can be modified to suit the required version styling.
`bump2version minor` increments the minor portion (middle number) by 1, and `bump2version major` changes the major portion or first digit of the version.

The bumpversion or bump2version (better maintained) documents can help- understand this clearly.

Please give me any suggestions you have so I can improve/change this. Thanks!

**Comment by nandahkrishna** [comment]
I guess I could change conf.py and setup.py to use the value rather than read from the cfg, if that's preferred. Should I go ahead and change it or is this fine?

**Comment by MainRo** [comment]
> I guess I could change conf.py and setup.py to use the value rather than read from the cfg, if that's preferred. Should I go ahead and change it or is this fine?

Let's see what @erikkemperman @dbrattli and @jcafhe think about this. Keep it as-is for now.

**Comment by dbrattli** [comment]
Merging this for now as I want to get the next beta out. If there are any issues then lets fail forward and fix the problem.

**Comment by dbrattli** [comment]
Note that I got a bug with this PR when doing the new beta release since the package name became 0.0.0 eg

```
-rw-r--r--  1 dbrattli  staff   183K May 20 07:44 Rx-0.0.0-py2.py3-none-any.whl
-rw-r--r--  1 dbrattli  staff    96K May 20 07:44 Rx-0.0.0.tar.gz
```

I think version needs to be set to build sucessfully. Fixed this by setting `version` = `current_version` in `setup.py`:

```py
version=project_meta.get("current_version"),
```

Could you @nandahkrishna verify that this fix is the preferred way?

**Comment by nandahkrishna** [comment]
Yes, as the .bumpversion.cfg file contains the version number, the line had to be added to setup.py - it is the right solution. Thanks for spotting it, slipped my mind.

**Pull request merged**